src/napari_epitools/_widget.py

```diff
@@ -369,11 +384,97 @@ def _create_cell_statistics_widgets() -> list[Widget]:
     return [image, labels, run, export]
 
 
+def _create_colour_labels_widgets() -> list[Widget]:
+    """Create widgets for colouring Labels layers by their featuers"""
+
+    create_colour_labels_heading = magicgui.widgets.create_widget(
+        label="<b>Colour by statistic</b>",  # bold label
+        widget_type="Label",
+        gui_only=True,
+    )
+
+    labels_tooltip = "Select a 'Labels' layer to set colourmap for"
+    labels = magicgui.widgets.create_widget(
+        annotation=napari.layers.Labels,
+        name="colourmap_input_labels",
+        label="labels",
+        options={"tooltip": labels_tooltip},
+    )
+
+    statisitc_tooltip = "Select which statistic to use for colouring the 'Labels'"
+    statistic = magicgui.widgets.create_widget(
+        name="colourmap_statistic",
+        value="None",
+        label="statistic",
+        widget_type="ComboBox",
+        options={
+            "choices": ["None", "area", "perimeter", "orientation", "neighbours"],
+            "tooltip": statisitc_tooltip,
+        },
+    )
+
+    lower_limit_tooltip = "Set the lower limit for the colourmap"
+    lower_limit = magicgui.widgets.create_widget(
+        value=0.0,
+        name="colourmap_lower_limit",
+        label="lower limit",
+        widget_type="FloatSpinBox",
+        options={
+            "tooltip": lower_limit_tooltip,
+            "min": -3.4 * 10**38,
+            "max": 3.4 * 10**38,
```

Comment:
Any reason for 10^38?

Reply:
by default `magicgui` sets an upper limit of 1000 to the float widgets (I'm not sure what the lower limit is). Some of the values in the statistics were larger than this, but I didn't really know what a reasonable upper limit would be for them. I [checked the maximum value](https://en.wikipedia.org/wiki/Single-precision_floating-point_format#:~:text=floating%2Dpoint%20variable%20has%20a%20maximum%20value%20of%20(2%20%E2%88%92%202%E2%88%9223)%20%C3%97%202127%20%E2%89%88%203.4028235%20%C3%97%201038) a `float32` can take whilst keeping [reasonable precision](https://en.wikipedia.org/wiki/Single-precision_floating-point_format#:~:text=This%20gives%20from%206%20to%209%20significant%20decimal%20digits%20precision.), and apparently it's roughly `3.4 x 10^38`!

Reply:
Fair enough haha!